jennchenn/uwexplore
# Create signup via email/password route

## Implementation description

- set up Firebase for auth [completes: BREAD-52]
- add `/auth/signup` route for adding new users to our application
- add /users?email=<>` route to fetch existing user info (will either end up removing this / scoping it for `ADMIN` role users only)
- updated user model to use `auth_id` as primary key for faster indexing
- still need to add in ability to signup via email



## Steps to test

1. Add Firebase credentials to local environment
2. Run: `curl -X POST -H "Content-Type: application/json" -d '{"name":"some bread", "email":"bread@verybread.com", "password": "somepassword"}' localhost:5000/auth/signup`
3. Verify 200 response received
4. Run: `curl "localhost:5000/users?email=bread@verybread.com"`
5. Verify appropriate user info is returned (should just be name, email and default `STUDENT` role)



## What should reviewers focus on?

- does it work?
